tools/xtce_generation/telemetry_generator.py

```diff
@@ -26,298 +64,280 @@ class TelemetryGenerator:
+        ccsds_container.attrib["abstract"] = "true"
+        ccsds_entry_list = Et.SubElement(ccsds_container, "xtce:EntryList")
 
         # Populate EntryList for CCSDSPacket SequenceContainer
-        for parameter_data in ccsds_parameters:
+        for parameter_data in _CCSDS_PARAMETERS:
             parameter_ref_entry = Et.SubElement(
-                ccsds_packet_entry_list, "xtce:ParameterRefEntry"
+                ccsds_entry_list, "xtce:ParameterRefEntry"
             )
-            parameter_ref_entry.attrib["parameterRef"] = parameter_data["name"]
-
-        # Create Packet SequenceContainer that use CCSDSPacket SequenceContainer
-        # as base container
-        science_container = Et.SubElement(container_set, "xtce:SequenceContainer")
-        science_container.attrib["name"] = container_name
+            name = parameter_data["name"]
 
-        base_container = Et.SubElement(science_container, "xtce:BaseContainer")
-        base_container.attrib["containerRef"] = "CCSDSPacket"
+            parameter_ref_entry.attrib["parameterRef"] = name
 
-        # Add RestrictionCriteria element to use the given APID for comparison
-        restriction_criteria = Et.SubElement(base_container, "xtce:RestrictionCriteria")
-        comparison = Et.SubElement(restriction_criteria, "xtce:Comparison")
-        comparison.attrib["parameterRef"] = "PKT_APID"
-        comparison.attrib["value"] = f"{self.apid}"
-        comparison.attrib["useCalibratedValue"] = "false"
+            # Add the parameter to the ParameterSet
+            parameter = Et.SubElement(self._parameter_set, "xtce:Parameter")
+            parameter.attrib["name"] = name
+            parameter.attrib["parameterTypeRef"] = name
 
-        # Populate EntryList for packet SequenceContainer
-        packet_entry_list = Et.SubElement(science_container, "xtce:EntryList")
-        parameter_refs = self.pkt.loc[7:, "mnemonic"].tolist()
+            description = Et.SubElement(parameter, "xtce:LongDescription")
+            description.text = parameter_data["description"]
 
-        for parameter_ref in parameter_refs:
-            parameter_ref_entry = Et.SubElement(
-                packet_entry_list, "xtce:ParameterRefEntry"
+            # Add the typeref to the parameter type set
+            parameter_type = Et.SubElement(
+                self._parameter_type_set, "xtce:IntegerParameterType"
             )
-            parameter_ref_entry.attrib["parameterRef"] = parameter_ref
+            parameter_type.attrib["name"] = name
+            parameter_type.attrib["signed"] = "false"
 
-        return telemetry_metadata
+            encoding = Et.SubElement(parameter_type, "xtce:IntegerDataEncoding")
+            encoding.attrib["sizeInBits"] = str(parameter_data["lengthInBits"])
+            encoding.attrib["encoding"] = "unsigned"
 
-    def create_remaining_parameters(self, parameter_set):
+    def _add_parameter_type(self, row):
```

Comment:
This is not important but slight personal preference. can you move this down with `_add_parameter()` function?

Reply:
Yep, I moved it down into the other section now.